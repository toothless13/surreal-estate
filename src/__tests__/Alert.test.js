import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Alert from "../components/Alert";

describe("Alert", () => {
  it("renders an error message", () => {
    const { getByText, asFragment } = render(<Alert message="Error!" />);

    expect(getByText(/Error/).textContent).toBe("Error!");
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a success message", () => {
    const { getByText, asFragment } = render(
      <Alert message="Success!!!" success />,
    );

    expect(getByText(/Success!!!/).textContent).toBe("Success!!!");
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders nothing if message is not truthy", () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();

    expect(alert).toMatchSnapshot();
  });
});
