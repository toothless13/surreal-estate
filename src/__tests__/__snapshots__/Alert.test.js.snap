// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders a success message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    Success!!!
  </div>
</DocumentFragment>
`;

exports[`Alert renders an error message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert renders nothing if message is not truthy 1`] = `<DocumentFragment />`;
